### PR TITLE
feat: add earn page and components

### LINKED ISSUE #309: UI: Earn Page

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

Added components for earn and routes

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #309 

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. etc...

## Screenshots (if applicable)
Screen Shot 2021-11-09 at 2 45 52 PM
Screen Shot 2021-11-09 at 2 45 56 PM

### CODE REVIEW

File: src/components/NestedMenu/NestedMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  return (
+    <>
+      {pathTo(route)
+        .filter(r => r.routes)
+        .map((r, index) => (
+          <Menu key={index} routes={r.routes} level={index} />
+        ))}
+    </>
+  )
```
Comment (0xdef1cafe) [inline src/components/NestedMenu/NestedMenu.tsx:67]:
```suggestion
  return pathTo(route)
        .filter(r => r.routes)
        .map((r, index) => (
          <Menu key={index} routes={r.routes} level={index} />
        ))
```

File: src/pages/Assets/AssetCards/StakingVaults/StakingVaults.tsx [hunk 2]
```diff
@@ -28,20 +28,25 @@ export const StakingVaults = ({ tokenId, isLoaded }: StakingVaultsProps) => {
...
-            </Button>
-          )}
-        </Card.Heading>
+      <Card.Header flexDir='row' display='flex'>
+        <Box>
+          <Card.Heading>
+            <Text translation={'assets.assetCards.stakingVaults'} />
+          </Card.Heading>
+          <RawText color='gray.500'>
+            Earn passive income by staking or deposting your assets into a vault.
```
Comment (0xdef1cafe) [inline src/pages/Assets/AssetCards/StakingVaults/StakingVaults.tsx:37]: translation

File: src/pages/Assets/AssetCards/StakingVaults/StakingVaults.tsx [hunk 3]
```diff
@@ -28,20 +28,25 @@ export const StakingVaults = ({ tokenId, isLoaded }: StakingVaultsProps) => {
...
+          </RawText>
+        </Box>
+        {VAULTS.length > 0 && (
+          <Button size='sm' ml='auto' variant='link' colorScheme='blue'>
+            See All <ArrowForwardIcon />
```
Comment (0xdef1cafe) [inline src/pages/Assets/AssetCards/StakingVaults/StakingVaults.tsx:42]: translation

File: src/pages/Earn/views/Farming.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Box } from '@chakra-ui/react'
+import { Page } from 'components/Layout/Page'
+import { RawText } from 'components/Text'
+
+export const Farming = () => {
+  return (
+    <Page>
+      <Box width='full'>
+        <RawText>Farming</RawText>
```
Comment (0xdef1cafe) [inline src/pages/Earn/views/Farming.tsx:9]: translation

File: src/pages/Earn/views/LiquidityPools.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Box } from '@chakra-ui/react'
+import { Page } from 'components/Layout/Page'
+import { RawText } from 'components/Text'
+
+export const LiquidityPools = () => {
+  return (
+    <Page>
+      <Box width='full'>
+        <RawText>Liquidity Pools</RawText>
```
Comment (0xdef1cafe) [inline src/pages/Earn/views/LiquidityPools.tsx:9]: translation

**Review by 0xdef1cafe — APPROVED** [review]

**Review by theoboldfrazier — APPROVED** [review]
👍

**Review by technojak — APPROVED** [review]

**Pull request merged**